### PR TITLE
more handle errors with multiple files

### PR DESCRIPTION
**Pull request by Ludmuterol**
added handling for case mentioned in (#4906) and added some tests as well.
But i have to say i stumbled upon something i need to ask about: is there a reason the more function returns an error with an exit code? cause when i compare:
```
~/Desktop/coreutils/target/debug$ ./coreutils more nonexistent_file
more: cannot open 'nonexistent_file': No such file or directory
~/Desktop/coreutils/target/debug$ echo $?
1
~/Desktop/coreutils/target/debug$ more nonexistent_file
more: cannot open nonexistent_file: No such file or directory
~/Desktop/coreutils/target/debug$ echo $?
0
```
is that an issue or smth that that doesn't need to be changed?
i tried to keep it this way but if multiple files are given it will always return no error. is this okay? or how should more handle one file failing and one working?

### CODE REVIEW

File: src/uu/more/src/more.rs [hunk 1]
```diff
@@ -105,29 +105,51 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
...
             let file = Path::new(file);
             if file.is_dir() {
-                terminal::disable_raw_mode().unwrap();
-                return Err(UUsageError::new(
-                    1,
-                    format!("{} is a directory.", file.quote()),
-                ));
+                if length == 1 {
+                    terminal::disable_raw_mode().unwrap();
+                    return Err(UUsageError::new(
```
Comment (sylvestre) [inline src/uu/more/src/more.rs:110]: could you please add tests to arrive in this error ? thanks

Comment (Ludmuterol) [reply src/uu/more/src/more.rs:110]: Sure i hope i didn't miss more. grcov didnt want to work last time i tried...

File: src/uu/more/src/more.rs [hunk 2]
```diff
@@ -105,29 +105,51 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
         while let (Some(file), next_file) = (files_iter.next(), files_iter.peek()) {
             let file = Path::new(file);
             if file.is_dir() {
-                terminal::disable_raw_mode().unwrap();
-                return Err(UUsageError::new(
-                    1,
-                    format!("{} is a directory.", file.quote()),
-                ));
+                if length == 1 {
```
Comment (tertsdiepraam) [inline src/uu/more/src/more.rs:108]: I'd like to suggest using the `show!` and `map_err_context` functionality:
- https://docs.rs/uucore/latest/uucore/macro.show.html
- https://docs.rs/uucore/latest/uucore/error/trait.FromIo.html

I think those would make it unnecessary to check the number of files and to log the error messages in the buffer.

**Review by tertsdiepraam — COMMENTED** [review]
Hi! I've been a little absent from this project for a while, so that's why I didn't see this sooner.

I like the idea here! We should definitely improve this case.

If I try this patch, I get the following:
```
::::::::::::::
oranda.json
::::::::::::::
more: 'src' is a directory.
::::::::::::::
oranda.json
::::::::::::::
{
  "project": {
    "name": "uutils coreutils"
...
```
That seems a bit strange and like there's something going on with the printing. I've made a comment with some functions that might help here.

`less` and util-linux `more` both print the error message and then wait for user input. This makes sense because the next file might obscure the message. However, for a first version we don't have to do that yet.

> But i have to say i stumbled upon something i need to ask about: is there a reason the more function returns an error with an exit code?

Nope, it'd be great if you could fix that! Thanks for finding that!

**Comment by Ludmuterol** [comment]
my terminal currently gives me the broken output i associate with the raw mode not being disabled. but i cant seem to find where the program could exit without disabling it first. and every test case i tried doesnt replicate the terminal not disabling the raw mode. that cargo test doesnt seem to execute the tests in a predictable order doesnt help with finding the case that leaves the terminal in raw mode... any keywords/ideas on how i can find the reason for that?

File: src/uu/more/src/more.rs (hunk 2 above)
Comment (Ludmuterol) [reply src/uu/more/src/more.rs]: well i was using the buffer so that i dont have to switch the raw mode for each message.. do i even have to do that, since the errormessages are printed to stderr and not stdout? sadly the  documentation for the raw mode doesnt mention stderr.. 
(https://docs.rs/crossterm/0.27.0/crossterm/terminal/index.html#raw-mode)

File: src/uu/more/src/more.rs [hunk 3]
```diff
@@ -105,25 +105,31 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
...
+                show!(USimpleError::new(
+                    0,
                     format!("cannot open {}: No such file or directory", file.quote()),
                 ));
+                terminal::enable_raw_mode().unwrap();
+                continue;
             }
             let opened_file = match File::open(file) {
                 Err(why) => {
                     terminal::disable_raw_mode().unwrap();
```
Comment (tertsdiepraam) [inline src/uu/more/src/more.rs:126]: I don't actually know why it stays in raw mode, but it seems unrelated to this change, so let's merge it!

**Review by tertsdiepraam — APPROVED** [review]
